cocotb/__init__.py

```diff
@@ -106,8 +107,36 @@ def _reopen_stream_with_buffering(stream_name):
 plusargs = {}
 """A dictionary of "plusargs" handed to the simulation."""
 
-# To save typing provide an alias to scheduler.add
-fork = scheduler.add
+
+def current_test() -> cocotb.decorators.RunningTest:
+    """Returns a reference to the currently running test"""
+    global scheduler
+    return scheduler._test
+
+
+def create_task(coro: Union[Coroutine, cocotb.decorators.RunningTask]) -> cocotb.decorators.RunningTask:
+    """
+    """
+    global scheduler
+    return scheduler.add(coro)
+
+
+def fork(coro: Union[Coroutine, cocotb.decorators.RunningTask]):
+    global scheduler
+
+    co = create_task(coro)
+
+    async def watcher():
+        try:
+            await co
+        except Exception as e:
+            if cocotb.triggers.Join(co) not in scheduler._trigger2coros:
+                current_test().abort(e)
```

Comment:
Does it make sense to warn the user that just because it worked for them once doesn't mean it will always work, and to inform them they may be looking for `create_task`?

Message lousy, but hopefully enough to give you the idea.
```suggestion
            if cocotb.triggers.Join(co) not in scheduler._trigger2coros:
                current_test().abort(e)
            else:
                warnings.warn("If you care about the result of fork, you might be after create_task instead")
```

Reply:
The 3.8 regression won't like that. Should we put a log there instead?